plugins/state_history_plugin/include/eosio/state_history_plugin/state_history_serialization.hpp

```diff
@@ -574,35 +566,43 @@ inline fc::optional<uint64_t> cap_error_code( const fc::optional<uint64_t>& erro
 
 template <typename ST>
 datastream<ST>& operator<<(datastream<ST>& ds, const history_context_wrapper<bool, eosio::chain::action_trace>& obj) {
+   const auto& trace = obj.obj;
+   if (!trace.return_value) {
+      fc::raw::pack( ds, fc::unsigned_int( 0 ));
+   } else {
+      fc::raw::pack( ds, fc::unsigned_int( 1 ));
+   }
    bool  debug_mode = obj.context;
    fc::raw::pack(ds, fc::unsigned_int(0));
```

Comment:
You only need one version field.

Reply:
Also, the version flipping creates a minefield for adding a _v2 in the future. It seems cleaner to always generate a _v1.

Reply:
Actually, it created a minefield when merging with another PR. _v1 in that branch includes `account_disk_deltas`, which doesn't exist in _v0.

Reply:
Here's how I resolved conflicts with that merge:

```
template <typename ST>
datastream<ST>& operator<<(datastream<ST>& ds, const history_context_wrapper<bool, eosio::chain::action_trace>& obj) {
   bool debug_mode = obj.context;
   fc::raw::pack(ds, fc::unsigned_int(1));
   fc::raw::pack(ds, as_type<fc::unsigned_int>(obj.obj.action_ordinal));
   fc::raw::pack(ds, as_type<fc::unsigned_int>(obj.obj.creator_action_ordinal));
   fc::raw::pack(ds, bool(obj.obj.receipt));
   if (obj.obj.receipt) {
      fc::raw::pack(ds, make_history_serial_wrapper(obj.db, as_type<eosio::chain::action_receipt>(*obj.obj.receipt)));
   }
   fc::raw::pack(ds, as_type<uint64_t>(obj.obj.receiver.to_uint64_t()));
   fc::raw::pack(ds, make_history_serial_wrapper(obj.db, as_type<eosio::chain::action>(obj.obj.act)));
   fc::raw::pack(ds, as_type<bool>(obj.obj.context_free));
   fc::raw::pack(ds, as_type<int64_t>(debug_mode ? obj.obj.elapsed.count() : 0));
   if (debug_mode)
      fc::raw::pack(ds, as_type<std::string>(obj.obj.console));
   else
      fc::raw::pack(ds, std::string{});
   history_serialize_container(ds, obj.db, as_type<flat_set<eosio::chain::account_delta>>(obj.obj.account_ram_deltas));
   history_serialize_container(ds, obj.db, as_type<flat_set<eosio::chain::account_delta>>(obj.obj.account_disk_deltas));

   fc::optional<std::string> e;
   if (obj.obj.except) {
      if (debug_mode)
         e = obj.obj.except->to_string();
      else
         e = "Y";
   }
   fc::raw::pack(ds, as_type<fc::optional<std::string>>(e));
   fc::raw::pack(ds,
                 as_type<fc::optional<uint64_t>>(debug_mode ? obj.obj.error_code : cap_error_code(obj.obj.error_code)));
   fc::raw::pack(ds, as_type<rc::optional<eosio::bytes>>(obj.obj.return_value));
   return ds;
}
```

Reply:
(I undid the `obj.obj => trace` since that made the merge more difficult)

Reply:
Gotcha, went back to the way it was and added the other fixes.